inference-engine/src/mkldnn_plugin/nodes/common/cpu_convert.cpp

```diff
@@ -46,73 +216,270 @@ struct ConvertContext {
+                break;
+            case Precision::I64:
+                range_0 = commonRange<src_t, PrecisionInfo<Precision::I64>::value_type>();
+                break;
+            case Precision::FP32:
+                range_0 = commonRange<src_t, PrecisionInfo<Precision::FP32>::value_type>();
+                break;
+            case Precision::FP16:
+                range_0 = commonRange<src_t, PrecisionInfo<Precision::FP16>::value_type>();
+                break;
+            case Precision::BF16:
+                range_0 = commonRange<src_t, PrecisionInfo<Precision::BF16>::value_type>();
+                break;
+            case Precision::FP64:
+                range_0 = commonRange<src_t, PrecisionInfo<Precision::FP64>::value_type>();
+                break;
+            case Precision::BOOL:
+                range_0 = commonRange<src_t, PrecisionInfo<Precision::BOOL>::value_type>();
+                break;
+            default:
+                IE_THROW() << "Unsupported precision";
+        }
+
+        auto range_1 = commonRange<src_t, dst_t>();
+
+        return std::make_pair(std::max(std::get<0>(range_0), std::get<0>(range_1)),
+                              std::min(std::get<1>(range_0), std::get<1>(range_0)));
+    }
+};
+
+template<typename T>
+struct TruncatePrecision;
+
+template<typename src_t, typename dst_t>
+struct TruncatePrecision<std::tuple<src_t, dst_t>> {
+    void operator()(TruncateContext & ctx) {
+        auto src = static_cast<const src_t *>(ctx.srcPtr);
+        auto dst = static_cast<dst_t *>(ctx.dstPtr);
+        src_t lbound, ubound;
+        std::tie(lbound, ubound) = ctx.range<src_t, dst_t>();
+
+        if (std::is_integral<src_t>::value
+            || ctx.interimPrc.is_float()
+            || std::is_integral<dst_t>::value) {
+            parallel_for(ctx.size, [&](size_t i) {
+                dst[i] = static_cast<dst_t>(std::max(std::min(src[i], ubound), lbound));
+            });
+        } else {
+            parallel_for(ctx.size, [&](size_t i) {
+                dst[i] = static_cast<dst_t>(std::trunc(std::max(std::min(src[i], ubound), lbound)));
+            });
+        }
+
+        ctx.converted = true;
+    }
+};
+
+bool isConversionTruncatesRange(const Precision & from, const Precision & to) {
+    return to.bitsSize() < from.bitsSize()
+            || (from.is_float() && !to.is_float())      // float -> integral
+            || (from.isSigned() != to.isSigned())       // signed <-> unsigned
+            || (to == Precision::BOOL && from != to);   // T -> bool
+}
+
 }   // namespace
 
 #define MKLDNN_CVT(ST, DT) OV_CASE2(Precision::ST, Precision::DT, PrecisionInfo<Precision::ST>::value_type, PrecisionInfo<Precision::DT>::value_type)
 
-void cpu_convert(const void *srcPtr, void *dstPtr, Precision srcPrc, Precision dstPrc, const size_t size) {
-    using namespace MKLDNNPlugin;
+#define MKLDNN_CVT_LIST                                                                             \
```

Comment:
I am not fully understand why we need to have to separate lists  MKLDNN_CVT_LIST and MKLDNN_TRUNC_LIST. Seems like they can be united.

Reply:
MKLDNN_TRUNC_LIST contains conversion pairs in which the source and target types are the same and is not required for conversions without intermediate precision. Ok, I'll include them to MKLDNN_CVT_LIST. It should work too.

Reply:
done